hooks/useCreateModal.tsx

```diff
@@ -3,18 +3,22 @@ import { useToast } from '@/components/ui/use-toast';
 import callPostApi from '@/lib/callPostApi';
 import isValidUrl from '@/lib/isValidUrl';
 import { useNeynarProvider } from '@/providers/NeynarProvider';
+import { useUi } from '@/providers/UiProvider';
 import { useRouter } from 'next/navigation';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
 const useCreateDialog = () => {
   const { signer } = useNeynarProvider();
   const [embedUrl, setEmbedUrl] = useState<string>('');
   const [channelId, setChannelId] = useState<string>();
-  const [isPostDialogOpen, setIsPostDialogOpen] = useState<boolean>(false);
+  const [selected, setSelected] = useState<number>();
+  const [isChannelListOpen, setIsChannelListOpen] = useState(false);
+  const { menuItems } = useUi();
   const { toast } = useToast();
   const router = useRouter();
+  const canCast = (typeof selected == 'number' && selected >= 0) || embedUrl;
 
-  const handlePost = () => {
+  const handlePost = (callback: () => void = () => {}) => {
```

Comment:
why are we passing a `callback` function???
- actual: fixing with additional code  
- expected: fixing this should be `simplifying` the code

less is more
simpel is best
small is beautiful